### DISCUSSION TITLE
[Question][Hw1][Problem 7] Questions on further assumptions of page rank algorithm

### DISCUSSION THREAD

**Discussion by yusungsim** [prosyslab-classroom/cs524-program-analysis#406] [Q&A]
Hi. I'd like to ask about more assumptions of the page rank algorithm of problem 7.

1. Can I assume that there will be no self-referencing pages? (ie. all pages doesn't not have link to themselves)

2. Can I assume that all `page_id` will appear in the `web`, through 1 ~ N?

  For example, there can be an example of `web` object like this one.
  ```ocaml
  let web_ex = [
    [(1, 3); (1, 4)];
    [(3, 4)];
  ]
  ```
  The maximum of `page_id` here is 4. However, in `web_ex`, 2 does not appear as a page.

  In this case, should we assume that page 2 does not exists, or page 2 exists and user will do random jump after the page 2?

Thank you in advance!

**Comment by hyunsukimsokcho** [comment]
1. No, there can be a self-referencing page.
2. No, although all pages should appear in the `web`, its id can sufficiently be random (as long as it is `Int`, it's fine).

Btw, your `web_ex` is wrongly typed. Please read the docs carefully.

**Comment by yusungsim** [comment]
Thank you! I also realized that `(page * page) list list !== (page * page list) list` just now... XD

**Comment by hyunsukimsokcho** [comment]
Closing.. please feel free to re-open if needed.
